### PR TITLE
Added `getTransaction` endpoints and missing Transaction models

### LINKED ISSUE #62: Implement provider method for `getTransaction` endpoint

### PR DESCRIPTION
**Pull request by bartekryba**
Closes #62

### CODE REVIEW

**Review by THenry14 — COMMENTED** [review]
looks good! only a few nitpicks

File: Sources/Starknet/Data/Transaction/Transaction.swift [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import Foundation
 
-public struct StarknetSequencerInvokeTransaction: StarknetSequencerTransaction, Equatable {
+public struct StarknetInvokeTransaction: StarknetTransaction {
```
[the remark this answers is not in the document]
Comment (bartekryba) [reply Sources/Starknet/Data/Transaction/Transaction.swift:3]: Well, that's actually a good question. Whether we should put version explicitly in the latest versions of transactions? How do you approach it in the jvm?

File: Sources/Starknet/Data/Transaction/SequencerTransaction.swift [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    public let calldata: StarknetCalldata
+
+    public let signature: StarknetSignature
+
+    public let maxFee: Felt
+
+    public let nonce: Felt
+
+    public let version: Felt
```
[the remark this answers is not in the document]
Comment (bartekryba) [reply Sources/Starknet/Data/Transaction/SequencerTransaction.swift:16]: I stopped putting version there explicitly, when I added the functionality to account to sign transaction for estimate fee. But it's only used in the SequencerTransaction, so I can set version explicitly in normal transactions

File: Sources/Starknet/Data/Transaction/Transaction.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (THenry14) [reply Sources/Starknet/Data/Transaction/Transaction.swift:3]: We put the versions in (at least in the transactions that have multiple versions supported by us), and use serializers to differentiate based on version var. I believe starknet.py have plans to include versions in the naming as well.

File: Sources/Starknet/Data/Transaction/SequencerTransaction.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (THenry14) [reply Sources/Starknet/Data/Transaction/SequencerTransaction.swift:16]: Sure, sounds good :+1:

File: Sources/Starknet/Data/Transaction/Transaction.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (bartekryba) [reply Sources/Starknet/Data/Transaction/Transaction.swift:3]: Sure, will do it this way as well. Also I already use version for deserializing as well

**Review by 4rgorok — APPROVED** [review]

File: Sources/Starknet/Data/Transaction/Transaction.swift [hunk 3]
```diff
@@ -194,20 +186,19 @@ public struct StarknetDeployAccountTransaction: StarknetTransaction {
...
     public let entrypointSelector: Felt
 
-    public let calldata: [Felt]
+    public let calldata: StarknetCalldata
 
     public let hash: Felt
 
-    public init(version: Felt, nonce: Felt, contractAddress: Felt, entrypointSelector: Felt, calldata: [Felt], hash: Felt) {
-        self.version = version
+    public init(nonce: Felt, contractAddress: Felt, entrypointSelector: Felt, calldata: [Felt], hash: Felt) {
```
Comment (THenry14) [inline Sources/Starknet/Data/Transaction/Transaction.swift:201]:
```suggestion
    public init(nonce: Felt, contractAddress: Felt, entrypointSelector: Felt, calldata: StarknetCalldata, hash: Felt) {
```

**Pull request merged**